### PR TITLE
Feature 33 project notifications

### PR DESCRIPTION
**Pull request by kaeline**
https://github.com/Metatavu/lipsanen-project-management-spec/issues/33

### CODE REVIEW

File: swagger.yaml [hunk 1]
```diff
@@ -1112,6 +1112,152 @@ paths:
...
+    delete:
+      operationId: deleteNotification
+      summary: Delete a notification
+      description: Delete a notification, only for testing purposes.
+      tags:
+        - Notifications
+      parameters:
+        - name: projectId
+          in: path
+          required: true
```
Comment (Daniilmass) [inline swagger.yaml]: Notification object itself doesn't have a link to a Project, why are we using a projectId as a parameter? Should it be a taskId? Or maybe even just notificationId is enough?

File: swagger.yaml [hunk 2]
```diff
@@ -1682,6 +1828,63 @@ components:
         - START_TO_START
         - FINISH_TO_FINISH
 
+    Notification:
+      type: object
+      description: Notification object that is created upon an event and sent to multiple users
+      required:
+        - message
+        - type
```
Comment (Daniilmass) [inline swagger.yaml:1851]: If notifications are task specific, should taskId be required? Or do you think it is fine as optional for future flexibility of having other notification types other than task related?

File: swagger.yaml [hunk 3]
```diff
@@ -1682,6 +1828,63 @@ components:
...
+        notification:
+          description: Full notification object
+          $ref: '#/components/schemas/Notification'
+        receiverId:
+          description: User ID that the notification was sent to
+          type: string
+          format: uuid
+        read:
+          description: Whether the notification has been read
+          type: boolean
```
Comment (Daniilmass) [inline swagger.yaml:1888]: We have a read parameter but how do we handle "dismissing" NotificationEvents i.e. if user presses the X (cross) button? This action should probably remove NotificationEvent from this user by either deleting or setting another status like "dismissed"

File: swagger.yaml (hunk 2 above)
Comment (kaeline) [reply swagger.yaml:1851]: I am not sure it's always applicable to all the types? If it's notification about task deletion for instance then we will have no valid task id at the point when notification is read

File: swagger.yaml (hunk 3 above)
Comment (kaeline) [reply swagger.yaml:1888]: Apparently I forgot the delete endpoint :D Added it now

File: swagger.yaml (hunk 1 above)
Comment (kaeline) [reply swagger.yaml]: It's for testing purposes only so you are right, just notificationId is good enough. This was leftover from when they were kept under "project". Fixed

**Review by lauriLukkarinen — APPROVED** [review]

**Review by Daniilmass — APPROVED** [review]

**Review by Daniilmass — APPROVED** [review]

File: swagger.yaml [hunk 4]
```diff
@@ -1681,6 +1843,62 @@ components:
         - START_TO_START
         - FINISH_TO_FINISH
 
+    Notification:
```
Comment (lauriLukkarinen) [inline swagger.yaml:1846]: Maybe add commentId also, so we can highlight the corresponding comment for the user, so we could give a direct link to the comment in the notifications list.

**Review by lauriLukkarinen — APPROVED** [review]

File: swagger.yaml (hunk 4 above)
Comment (kaeline) [reply swagger.yaml:1846]: In the next PR that introduces the comments :+1:

**Pull request merged**